gpu/octree/src/cuda/knn_search.cu

```diff
@@ -222,87 +222,50 @@ namespace pcl { namespace device { namespace knn_search
         }
 
         template<int CTA_SIZE>
-		__device__ __forceinline__ int NearestWarpKernel(const float* points, int points_step, int length, const float3& active_query, float& dist)
+		__device__ __forceinline__ int NearestWarpKernel(const int beg, const int points_step, const int length, const float3& active_query, float& dist)
 		{                        						
-            __shared__ volatile float dist2[CTA_SIZE];
-            __shared__ volatile int   index[CTA_SIZE];
-			
-            int tid = threadIdx.x;
-			dist2[tid] = std::numeric_limits<float>::max();
+			int index = 0;
+			float dist2 = std::numeric_limits<float>::max();
 
 			//serial step
             for (int idx = Warp::laneId(); idx < length; idx += Warp::STRIDE)
 			{
-				float dx = points[idx                  ] - active_query.x;
-				float dy = points[idx + points_step    ] - active_query.y;
-				float dz = points[idx + points_step * 2] - active_query.z;
+				const float dx = batch.points[beg + idx                  ] - active_query.x;
+				const float dy = batch.points[beg + idx + points_step    ] - active_query.y;
+				const float dz = batch.points[beg + idx + points_step * 2] - active_query.z;
 
-				float d2 = dx * dx + dy * dy + dz * dz;
+				const float d2 = dx * dx + dy * dy + dz * dz;
 
-				if (dist2[tid] > d2)
+				if (dist2 > d2)
 				{
-					dist2[tid] = d2;
-					index[tid] = idx;                            
+					dist2 = d2;
+					index = idx;
 				}
 			}
-			//parallel step
-			unsigned int lane = Warp::laneId();
 
-			float mind2 = dist2[tid];
+			//find minimum distance among warp threads
+            constexpr unsigned FULL_MASK = 0xFFFFFFFF;
+            static_assert(sizeof(KernelPolicy::WARP_SIZE) <= sizeof(unsigned int));
```

Comment:
this check is wrong. `WARP_SIZE` represents the number of bits in the mask. `sizeof` gives you the number of **bytes**. you also need an error message here explaining the user what is wrong if the static_assert is violated.

```suggestion
            static_assert(KernelPolicy::WARP_SIZE <= 8*sizeof(unsigned int));
```